### PR TITLE
feat: add batch array tools for array property operations

### PR DESCRIPTION
**Pull request by kzmshx**
## Summary

- Add 4 new MCP tools for batch operations on array properties:
  - `batch_array_add`: Add values to array properties
  - `batch_array_remove`: Remove values from array properties
  - `batch_array_replace`: Replace values in array properties
  - `batch_array_sort`: Sort array properties

- Each tool supports glob patterns and handles edge cases:
  - Missing properties (create on add, skip on others)
  - Non-array properties (warn and skip)
  - Duplicate detection (configurable for add)
  - Already-sorted arrays (skip on sort)

## Test plan

- [x] All existing tests pass
- [x] New tests for all 4 tools (21 test cases)
- [x] Linter passes

### CODE REVIEW

File: src/frontmatter_mcp/server.py [hunk 1]
```diff
@@ -173,6 +173,311 @@ def batch_update(
...
+            warnings.append(f"Failed to update {rel_path}: {e}")
+
+    response: dict[str, Any] = {
+        "updated_count": len(updated_files),
+        "updated_files": updated_files,
+    }
+    if warnings:
+        response["warnings"] = warnings
+
+    return response
```
[the remark this answers is not in the document]
Comment (kzmshx) [reply src/frontmatter_mcp/server.py]: Acknowledged. Keeping current structure for now because:
1. Each function has slightly different logic
2. Duplication is contained within this file
3. Helper extraction adds indirection without significant benefit at this scale

Will refactor if more batch array operations are added.
